src/common/summarize-profile.js

```diff
@@ -263,17 +276,17 @@ function countStacksInCategory(profile, summaries, category = 'uncategorized') {
  * @param {array} profile - The current profile.
  * @returns {array} Stacks mapped to categories.
  */
-export function categorizeThreadSamples(profile) {
+export function categorizeThreadSamples(profile: Profile): Array<string[]> {
   return timeCode('categorizeThreadSamples', () => {
-    const summaries = profile.threads.map(thread => {
+    const summaries:Array<string[]> = profile.threads.map(thread => {
```

Comment:
style: perhaps move this into a helper `mapProfileToSummaries` with a return type.